packages/vscode-extension/src/extension.ts

```diff
@@ -2,31 +1,283 @@
+                    progress.report({ increment: ++done / allDependencies.length * 100 })
+                })
+            )
+        })
+        return Promise.resolve(thenable)
+    }
+
+    async openPackage($package: Package) {
+        await vscode.commands.executeCommand(
+            'revealInExplorer',
+            vscode.Uri.parse($package.directory)
+        )
+    }
+
+    async openDependency(dependency: Dependency) {
+        if (!dependency.sourceDirectory) {
+            await vscode.window.showErrorMessage("Source code is not downloaded")
+            return
+        }
+        const directoryUri = vscode.Uri.file(dependency.sourceDirectory)
+        const fileUri = vscode.Uri.joinPath(directoryUri, 'README.md')
+        // TODO: If does not exist, try with 'readme' from package.json, 'src'/'main' from package.json, package.json itself, or else fall back to URL
+        await vscode.commands.executeCommand('markdown.showPreview', fileUri)
+    }
+
+    async openDependencyFolder(dependency: Dependency, relativePath: string) {
+        if (!dependency.sourceDirectory) {
+            await vscode.window.showErrorMessage("Source code is not downloaded")
+            return
+        }
+        const directoryUri = vscode.Uri.file(dependency.sourceDirectory)
+        const fileUri = vscode.Uri.joinPath(directoryUri, relativePath)
+        await vscode.commands.executeCommand('vscode.open', fileUri)
+    }
+
+    async openReference(reference: Reference) {
+        if (!reference.dependency.sourceDirectory) {
+            await vscode.window.showErrorMessage("Source code is not downloaded")
+            return
+        }
+        const directoryUri = vscode.Uri.file(reference.dependency.sourceDirectory)
+        const fileUri = vscode.Uri.joinPath(directoryUri, reference.file)
+        const document = await vscode.workspace.openTextDocument(fileUri)
+        const position = new vscode.Position(reference.position.row - 1, (reference.position.column ?? 1) - 1)
+        await vscode.window.showTextDocument(document, {
+            preview: true,
+            selection: new vscode.Selection(position, position)
+        })
+    }
+
+    async refreshDependencies($package: Package, cancellationToken?: vscode.CancellationToken) {
+        await $package.updateDependencies(this.dowdep, async () => {
+            if (cancellationToken?.isCancellationRequested) {
+                throw new vscode.CancellationError()
+            }
+            this.notifyModelObservers()
+        })
+    }
+
+    async refreshReferences(dependency: Dependency, cancellationToken?: vscode.CancellationToken) {
+        await dependency.updateReferences(this.dowdep, async () => {
+            if (cancellationToken?.isCancellationRequested) {
+                throw new vscode.CancellationError()
+            }
+            this.notifyModelObservers()
+        })
+    }
+
+    private wrapWithLogger<TIn extends any[], TOut>(fun: (...args: TIn) => Promise<TOut>) {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/LinqLover/downstream-repository-mining/pullRequest?prid=8058375)
```suggestion
    private wrapWithLogger<TIn extends unknown[], TOut>(fun: (...args: TIn) => Promise<TOut>) {
```